### PR TITLE
Follow-up on #469 for Linux this time, overriding read only pages cod…

### PR DESCRIPTION
**Pull request by devnexen**
…e path too.

### CODE REVIEW

File: src/pal/pal_linux.h [hunk 1]
```diff
@@ -68,20 +68,40 @@ namespace snmalloc
...
+    }
+
+    /**
+     * Notify platform that we will be using these pages.
+     */
+    template<ZeroMem zero_mem>
+    static void notify_using(void* p, size_t size) noexcept
+    {
+      PALPOSIX<PALLinux>::notify_using<zero_mem>(p, size);
+      madvise(p, size, MADV_DONTDUMP);
```
Comment (mjp41) [inline src/pal/pal_linux.h:104]: DODUMP?

Comment (devnexen) [reply src/pal/pal_linux.h:104]: ah yes copy/paste sorry :)

**Review by davidchisnall — APPROVED** [review]
Looks good to me, is it marking memory returned by `reserve` as not being included in core dumps?

**Comment by mjp41** [comment]
> Looks good to me, is it marking memory returned by `reserve` as not being included in core dumps?

I think Linux requires a separate call to `madvise`, there is no equivalent of `MAP_NOCORE` that I could see.

**Comment by devnexen** [comment]
> > Looks good to me, is it marking memory returned by `reserve` as not being included in core dumps?
> 
> I think Linux requires a separate call to `madvise`, there is no equivalent of `MAP_NOCORE` that I could see.

thanks good catch I missed that one.

**Comment by devnexen** [comment]
> > Looks good to me, is it marking memory returned by `reserve` as not being included in core dumps?
> 
> I think Linux requires a separate call to `madvise`, there is no equivalent of `MAP_NOCORE` that I could see.

makes me think ... openbsd is literally the other way around ..

**Comment by devnexen** [comment]
Do you know if windows can too ? It seems to me it is not possible looking at virtualloc2* pages

**Pull request merged**